src/v/redpanda/admin_server.cc

```diff
@@ -670,6 +670,62 @@ static json_validator make_cluster_config_validator() {
     return json_validator(schema);
 }
 
+/**
+ * This function provides special case validation for configuration
+ * properties that need to check other properties' values as well
+ * as their own.
+ *
+ * Ideally this would be built into the config_store/property generic
+ * interfaces, but that's a lot of plumbing for a few relatively simple
+ * checks, so for the moment we just do the checks here by hand.
+ */
+void config_multi_property_validation(
+  ss::sstring const& username,
+  cluster::config_update_request const& req,
+  config::configuration const& updated_config,
+  std::map<ss::sstring, ss::sstring>& errors) {
+    absl::flat_hash_set<ss::sstring> modified_keys;
+    for (const auto& i : req.upsert) {
+        modified_keys.insert(i.first);
+    }
+
+    bool auth_now_enabled = config::shard_local_cfg().admin_api_require_auth();
+    if (modified_keys.contains("admin_api_require_auth")) {
+        auth_now_enabled = updated_config.admin_api_require_auth();
+    }
+
+    if (
+      (modified_keys.contains("admin_api_require_auth")
+       || modified_keys.contains("superusers"))
+      && auth_now_enabled) {
+        // We are switching on admin_api_require_auth.  Apply rules to prevent
+        // the user "locking themselves out of the house".
+
+        // There must be some superusers defined
+        auto superusers = config::shard_local_cfg().superusers();
+        if (modified_keys.contains("superusers")) {
+            superusers = updated_config.superusers();
+        }
+
+        const bool auth_was_enabled
+          = config::shard_local_cfg().admin_api_require_auth();
+
+        absl::flat_hash_set<ss::sstring> superusers_set(
+          superusers.begin(), superusers.end());
+        if (superusers.empty()) {
+            // Some superusers must be defined, or nobody will be able
+            // to use the admin API after this request.
+            errors["admin_api_require_auth"] = "No superusers defined";
+            return;
+        } else if (!superusers_set.contains(username) && !auth_was_enabled) {
```

Comment:
should the rule be that _some_ superuser exists, not that it has to be the current user? since auth was disabled, it could be something like user john creates the admin user and then as the final step enables auth?

Reply:
We could allow that, but it creates the risk that John got it wrong and put the wrong user in `superusers`, or that he forgot the password he just used to create the admin user.  This way, we are forcing that the client proves they have a password  that will really let them back in.

Reply:
that makes sense

Reply:
I guess this can be taken further and not allow a user to be deleted if it is the last super user or something. Probably more cases than that, but as a guard rail this makes sense!